blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -274,9 +288,29 @@ private static Granularity granularityFromPointsGeometric(double requestedDurati
                 break;
             }
         }
+
+        if (gran == null) {
+            gran = Granularity.LAST;
+        }
+
         return gran;
     }
 
+    private static long getTTL(String tenantid, Granularity g) {
+        long ttl;
+        try {
+            ttl = (long)TTL_PROVIDER.getTTL(
```

Comment:
Also this is a useful utility method than @chinmay-gupte is going to want for rollupsOnRead().  So it probably go somewhere else, maybe one of the ttl classes.